### DISCUSSION TITLE
Improve REST Connector global and query heading bindings+

### DISCUSSION THREAD

**Discussion by chrisshaw** [Budibase/budibase#2726] [Ideas]
> Pulled over from https://github.com/Budibase/budibase/discussions/2685

In general, the REST API data source is a bit tedious, but generally fine.

The absolute biggest issue is [this one](https://github.com/Budibase/budibase/discussions/2685), with the (authorization) header bindings. My ideal developer experience would be to insert into the REST connector's global headers something like {{ Current User.oidc.token }} and be done.

---

Other things:

1. I keep bothering you about GraphQL :wink: . I'm biting the bullet and just sticking the queries and mutations into REST, but really I'd prefer an experience closer to Insomnia's: I can (a) tell the query it's a GraphQL query, (b) I can load the schema, (c) view the docs, and (d) type queries with autocomplete. Tooljet does this with a GraphQL data source and embedding GraphiQL.
2. We don't use the OpenAPI spec but I'm sure others would like auto-populated queries from such definitions. And if we **did** have that, I could use something like [sofa](https://www.sofa-api.com/docs/recipes/open-api) to generate one from our GraphQL schema and use that as a workaround.
3. Cherry on top would be integration with Automations for complex chains. Of course you could do this at the other end of the endpoint, but then that doesn't sound very low-code. (This is maybe just a separate source. The connection was that you could hit your own webhook Automation trigger today and so this is simply making that explicit.

**Comment by mjashanks** [comment]
Thanks Chris! Linking this too: https://github.com/Budibase/budibase/discussions/2727

**Reply by chrisshaw** [reply]
:+1: I did edit my answer with a super nice-to-have: using Automations as a data source.
